trains.js

```diff
@@ -11,25 +13,31 @@ var RELEASE_CALENDAR = "mozilla.com_2d37383433353432352d3939@resource.calendar.g
 
 var versions = {};
 
-function getVersion(responseText, keyJSON) {
-  contentJSON = JSON.parse(responseText);
-  return contentJSON[keyJSON].replace(/[ab]\d+$/, '').replace(/esr$/, '');
+function getVersion(jsonContent, keyJSON) {
+  return jsonContent[keyJSON].replace(/[ab]\d+$/, '').replace(/esr$/, '');
 }
 
-function fetchData(branch, keyJSON, description) {
+function fetchData(jsonContent, branch, keyJSON, description) {
+  if (Object.keys(jsonContent).length > 0 && jsonContent.constructor === Object) {
```

Comment:
This isn't quite the way I would have written it, but I'll just do a fixup commit after landing this vs. holding your PR up longer.